### PR TITLE
feat(rpc-server): Support block parameter

### PR DESCRIPTION
**Pull request by classicalliu**

### CODE REVIEW

**Review by jjyr — CHANGES_REQUESTED** [review]

File: crates/rpc-server/src/registry.rs [hunk 1]
```diff
@@ -209,15 +210,41 @@ async fn execute_l2transaction(
...
 
-    let raw_block = store.get_tip_block()?.raw();
+    let db = store.begin_transaction();
+    let block_number = match block_number {
+        Some(num) => num.value(),
+        None => db.get_tip_block()?.raw().number().unpack(),
+    };
+    let block_hash = match db.get_block_hash_by_number(block_number)? {
+        Some(block_hash) => block_hash,
+        None => return Ok(None),
```
[the remark this answers is not in the document]
Comment (classicalliu) [reply crates/rpc-server/src/registry.rs]: infura `eth_call`

```json
{
    "jsonrpc": "2.0",
    "id": 2,
    "error": {
        "code": -32000,
        "message": "header not found"
    }
}
```

godwoken `execute_raw_l2transaction` now:

```json
{
    "jsonrpc": "2.0",
    "error": {
        "code": 0,
        "message": "header not found"
    },
    "id": 1
}
```

And another question is `get_balance` throws db error when `block_number` to big, should we unify this?

```json
{
    "jsonrpc": "2.0",
    "error": {
        "code": 0,
        "message": "DB error can't find block hash"
    },
    "id": 1
}
```

[the remark this answers is not in the document]
Comment (jjyr) [reply crates/rpc-server/src/registry.rs]: We should return a special code to represent block is not found, and the web3 server should check the returned code and return a ethereum-compatible response to users.

[the remark this answers is not in the document]
Comment (jjyr) [reply crates/rpc-server/src/registry.rs]: Return this error to specific a response code https://docs.rs/jsonrpc-v2/0.10.1/jsonrpc_v2/enum.Error.html

[the remark this answers is not in the document]
Comment (classicalliu) [reply crates/rpc-server/src/registry.rs]:
```json
{
    "jsonrpc": "2.0",
    "error": {
        "code": -32000,
        "message": "header not found"
    },
    "id": 1
}
```

All RPCs which has block parameter param will check and throw this error when `block_number` bigger than tip.

**Review by jjyr — APPROVED** [review]

**Pull request merged**